srfi-sync.rkt

```diff
@@ -0,0 +1,631 @@
+     ""
+     "Replaces SRFI standard documents and auxiliary files by checking"
+     "out the upstream Git repositories and applying Racket-specific"
+     "transformations to the HTML files."
+     ""
+     "Existing files will be OVERWRITTEN as needed with NO WARNING."
+     #:help-labels
+     "---------------------------- actions ------------------------------"
+     "     (If no action is provided, this script exits immediately.)"
+     #:once-any
+     [("--sync") "Checkout and rewrite"
+      (set! checkout? #t)
+      (set! rewrite? #t)]
+     [("--checkout") "Checkout; don't rewrite"
+      (set! checkout? #t)]
+     [("--rewrite") "Rewrite; don't checkout (needs --work-dir)"
+      (set! rewrite? #t)]
+     #:help-labels
+     "------------------------- configuration ---------------------------"
+     #:once-each
+     [("--work-dir") path "Keep upstream files in <path> (default: temporary)"
+      (set! work-dir path)]
+     [("--owner") github-owner "Use instead of \"scheme-requests-for-implementation\""
+      (current-owner github-owner)]
+     [("--ref") git-ref "Use instead of 'head"
+      (current-ref git-ref)]
+     #:help-labels
+     "------------------------------ misc -------------------------------"
+     #:args ()
+     (if (or checkout? rewrite?)
+         (srfi-sync-all #:work-dir work-dir
+                        #:checkout? checkout?
+                        #:rewrite? rewrite?)
+         (eprintf "warning: no action specified\n")))))
+
+(require xml
+         net/url-string
+         net/git-checkout
+         "srfi-doc/srfi/scribblings/util.rkt"
+         racket/runtime-path)
+
+(define current-owner
+  (make-parameter "scheme-requests-for-implementation"))
+(define current-ref
+  (make-parameter 'head))
+
+(define (srfi-sync-all #:work-dir [work-dir #f]
+                       #:checkout? [checkout? #t]
+                       #:rewrite? [rewrite? #t]
+                       #:owner [owner (current-owner)]
+                       #:ref [ref (current-ref)])
+  (when rewrite?
+    (unless (or checkout? work-dir)
+      (error 'srfi-sync-all "rewrite without checkout requires work-dir")))
+  (parameterize ([current-owner owner]
+                 [current-ref ref])
+    (define ((go tmp))
+      (when checkout?
+        (srfi-checkout-all tmp))
+      (when rewrite?
+        (srfi-rewrite-all tmp)))
+    (cond
+      [work-dir
+       (make-directory* work-dir)
+       ((go work-dir))]
+      [else
+       (define tmp (make-temporary-file "tmp-srfi-sync~a" 'directory))
+       (dynamic-wind void
+                     (λ ()
+                       (call-with-continuation-barrier (go tmp)))
+                     (λ ()
+                       (delete-directory/files tmp)))])))
+
+(define the-srfis
+  `(,@'() 1 2 4 5 6 7 8 9
+          11 13 14 16 17 19
+          23 25 26 27 28 29
+          30 31 34 35 38 39
+          40 41 42 43 45 48
+          54 57 59
+          60 61 62 63 64 66 #;67 69 ;; skip 67 for now
```

Comment:
I didn't address SRFI 67 because I'm unclear on the relationship between Racket's copy and the upstream one in <https://github.com/scheme-requests-for-implementation/srfi-67>. The upstream copy is dated August 18, 2005, while Racket's is dated December 3, 2005. [Racket's copy](https://docs.racket-lang.org/srfi/srfi-std/srfi-67/srfi-67.html#node_index_start) also has just a `[Go to index]` at the top right linking to the bottom of the same page, which seems nicer than [srfi.schemers.org](https://srfi.schemers.org/srfi-67/srfi-67.html), where they have `[Go to first, previous, next page;   contents;   index]`, even though there are only two pages: the content and the index. On the other hand, [one section of Racket's copy](https://docs.racket-lang.org/srfi/srfi-std/srfi-67/srfi-67.html#node_sec_5) "was removed … due to the math." By my count it's 795 KiB of math (as `.png` files), which seems like a potentially reasonable price to pay to stay closer to upstream.

The HTML files declare that they were https://github.com/scheme-requests-for-implementation/srfi-67/blob/abbc48dfbcf5dafb704ea76a7af7eec4c7608663/srfi-67.html#L5-L8 and https://github.com/racket/srfi/blob/e32d29b30e91f392a391cfc1fdcb9edfda9b4883/srfi-doc/srfi/scribblings/srfi-std/srfi-67/srfi-67.html#L5-L8 but neither this repository nor <https://github.com/scheme-requests-for-implementation/srfi-67> has `srfi-67.tex`, though upstream does have `.dvi`, `.ps`, and `.pdf`. So I'm not sure if Racket should take the upstream version, if we should contribute Racket's single-page HTML version upstream, or if  there's a TeX source somewhere we should be regenerating all of this from.

@soegaard, I see that you were a co-author of SRFI 67: could you shed any light on these differences?